src/Nethermind/Nethermind.Core/Extensions/Bytes.cs

```diff
@@ -546,6 +564,37 @@ public static string ByteArrayToHexViaLookup32Safe(byte[] bytes, bool withZeroX)
             int length = bytes.Length * 2 + (withZeroX ? 2 : 0);
             StateSmall stateToPass = new(bytes, withZeroX);
 
+            return string.Create(length, stateToPass, (chars, state) =>
```

Comment:
QQ: does it make any difference if you use static lambda, or the CLR already optimize it away?

Reply:
Runtime already optimizes, but static ensures you aren't accidently capturing any variables so causing an allocation each time its called.

Have added